### PR TITLE
[SPARK-27422][SQL] current_date() should return current date in the session time zone

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

In the PR, I propose to revert 2 commits https://github.com/apache/spark/commit/06abd06112965cd73417ccceacdbd94b6b3d2793 and https://github.com/apache/spark/commit/61561c1c2d4e47191fdfe9bf3539a3db29e89fa9, and take current date via `LocalDate.now` in the session time zone. The result is stored as days since epoch `1970-01-01`.

## How was this patch tested?

It was tested by `DateExpressionsSuite`, `DateFunctionsSuite`, `DateTimeUtilsSuite`, and `ComputeCurrentTimeSuite`.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan Please, take a look at the PR.

File: docs/sql-migration-guide-upgrade.md [hunk 1]
```diff
@@ -118,9 +118,7 @@ license: |
...
 
-  - In Spark version 2.4 and earlier, when reading a Hive Serde table with Spark native data sources(parquet/orc), Spark will infer the actual file schema and update the table schema in metastore. Since Spark 3.0, Spark doesn't infer the schema anymore. This should not cause any problems to end users, but if it does, please set `spark.sql.hive.caseSensitiveInferenceMode` to `INFER_AND_SAVE`.
-
-  - In Spark version 2.4 and earlier, the `current_date` function returns the current date shifted according to the SQL config `spark.sql.session.timeZone`. Since Spark 3.0, the function always returns the current date in the `UTC` time zone.
+  - In Spark version 2.4 abd earlier, when reading a Hive Serde table with Spark native data sources(parquet/orc), Spark will infer the actual file schema and update the table schema in metastore. Since Spark 3.0, Spark doesn't infer the schema anymore. This should not cause any problems to end users, but if it does, please set `spark.sql.hive.caseSensitiveInferenceMode` to `INFER_AND_SAVE`.
```
Comment (cloud-fan) [inline docs/sql-migration-guide-upgrade.md]: can we keep this typo fix?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -2572,7 +2572,7 @@ object functions {
   }
 
   /**
-   * Returns the current date in the UTC time zone as a date column.
+   * Returns the current date as a date column.
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: do we need to change the python and R functions?

**Comment by cloud-fan** [comment]
LGTM. Date type in Spark is timezone independent and I think this behavior fits better.

**Comment by cloud-fan** [comment]
Note that, `current_timestamp` doesn't need to know the timzone, because timestamp type in Spark represents an absolute point in time.

**Review by HyukjinKwon — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 2 above)
Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: No, the PR https://github.com/apache/spark/pull/24185 did touch them. So, comments for the functions don't mention UTC time zone.

File: docs/sql-migration-guide-upgrade.md (hunk 1 above)
Comment (MaxGekk) [reply docs/sql-migration-guide-upgrade.md]: fixed

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**